File: java/1010_PairOfSongsWithTotalDurations.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int[] moddedTimes = new int[60];
        int count = 0;
        for (int t : time) moddedTimes[t % 60] += 1;
        for (int t = 0; t <= 30; ++t) {
            if (t == 0 || t == 30) {
                for (int x = 2; x <= moddedTimes[t]; ++x) {
                    count += (x - 1);
                }
            } else {
                count += (moddedTimes[t] * moddedTimes[60 - t]);
            }
        }
        return count;
    }
}

File: swift/1010_PairOfSongsWithTotalDurations.swift
class Solution {
    func numPairsDivisibleBy60(_ time: [Int]) -> Int {
        var moddedTimes = [Int](repeating: 0, count: 60)
        var count = 0
        
        for t in time { moddedTimes[t % 60] += 1 }
        for t in 0...30 {
            if t == 0 || t == 30 {
                if moddedTimes[t] < 2 { continue }
                for x in 1..<moddedTimes[t] {
                    count += x
                }
            } else {
                count += (moddedTimes[t] * moddedTimes[60 - t])
            }
        }
        return count
    }
}
